File: src/liveplugin/MakePluginFilesAlwaysEditable.java
package liveplugin;

import com.intellij.openapi.fileEditor.impl.NonProjectFileWritingAccessExtension;
import com.intellij.openapi.util.io.FileUtil;
import com.intellij.openapi.vfs.VirtualFile;
import org.jetbrains.annotations.NotNull;

public class MakePluginFilesAlwaysEditable implements NonProjectFileWritingAccessExtension {
    @Override public boolean isWritable(@NotNull VirtualFile file) {
        return FileUtil.startsWith(file.getPath(), LivePluginAppComponent.pluginsRootPath());
    }
}

File: src/main/liveplugin/MakePluginFilesAlwaysEditable.kt
package liveplugin

import com.intellij.openapi.fileEditor.impl.NonProjectFileWritingAccessExtension
import com.intellij.openapi.util.io.FileUtil
import com.intellij.openapi.vfs.VirtualFile

class MakePluginFilesAlwaysEditable: NonProjectFileWritingAccessExtension {
    override fun isWritable(file: VirtualFile) =
        FileUtil.startsWith(file.path, LivePluginAppComponent.livePluginsPath)
}
